scripts/02_reporting/tables.R

```diff
@@ -346,46 +473,168 @@ tab_cost_est_phase1 <- dplyr::left_join(
     `Habitat value` = habitat_value,
     `Stream Width (m)` = downstream_channel_width_meters,
     Fix = crossing_fix_code,
-    `Cost Est ( $K)` =  cost_est_1000s,
+    `Cost Est ( $K)` = cost_est_1000s,
     `Habitat Upstream (km)` = st_network_km,
     `Cost Benefit (m / $K)` = cost_gross,
-    `Cost Benefit (m2 / $K)` = cost_area_gross)
+    `Cost Benefit (m2 / $K)` = cost_area_gross
+  ) |>
+  dplyr::filter(!source == "pscis_phase2.xlsm") |>
+  dplyr::select(-source)
 
 
 
+## Phase 2 ------------------------------
 
-# Phase 2 overview table ------------------------------------------
+# Now prepare phase 2 cost estimates.
 
-# Overview of habitat confirmation sites used in the results section
+# Step 1: Join habitat confirmation priorities data to upstream habitat length
+tab_cost_est_prep7 <- dplyr::left_join(
+  tab_cost_est_prep4,
+  dplyr::select(
+    dplyr::filter(habitat_confirmations_priorities, location == 'us'),
+    site,
+    upstream_habitat_length_m
+  ),
+  by = c('pscis_crossing_id' = 'site')
+) |>
+  dplyr::mutate(
+    cost_net = round(upstream_habitat_length_m * 1000 / cost_est_1000s, 1),
+    cost_area_net = round((upstream_habitat_length_m * 1000 * downstream_channel_width_meters * 0.5) / cost_est_1000s, 1) ## this is a triangle area!
+  )
+
+# Step 2: Join habitat site data for average channel width using stringr for pattern matching
+tab_cost_est_prep8 <- dplyr::left_join(
+  tab_cost_est_prep7,
+  dplyr::select(
+    hab_site |>
+      dplyr::filter(
+        !stringr::str_detect(local_name, 'ds') &
+          !stringr::str_detect(local_name, 'ef') &
+          !stringr::str_detect(local_name, '\\d$')
+      ),
+    site, avg_channel_width_m
+  ),
+  by = c('pscis_crossing_id' = 'site')
+)
+
+
+# Step 3: Filter and select relevant columns for Phase 2 cost estimates
+tab_cost_est_prep9 <- tab_cost_est_prep8 |>
+  dplyr::filter(source == "pscis_phase2.xlsm") |>
+  dplyr::select(
+    pscis_crossing_id,
+    stream_name,
+    road_name,
+    barrier_result,
+    habitat_value,
+    avg_channel_width_m,
+    crossing_fix_code,
+    cost_est_1000s,
+    upstream_habitat_length_m,
+    cost_net,
+    cost_area_net,
+    source
+  ) |>
+  dplyr::mutate(upstream_habitat_length_m = round(upstream_habitat_length_m, 0))
 
-tab_overview_prep1 <- form_pscis|>
-  dplyr::filter(assess_type_phase2 == "Yes") |>
-  dplyr::select(pscis_crossing_id, stream_name, road_name, road_tenure, easting, northing, utm_zone, habitat_value)
+# Step 4: Prepare the Phase 2 cost estimates for the table
+tab_cost_est_phase2 <- tab_cost_est_prep9 |>
+  dplyr::arrange(pscis_crossing_id) |>
+  dplyr::select(-source)
 
-tab_overview_prep2 <- habitat_confirmations_priorities|>
-  dplyr::filter(location == 'us')|>
-  dplyr::select(site, species_codes, upstream_habitat_length_m, priority, comments)|>
-  dplyr::mutate(upstream_habitat_length_km = round(upstream_habitat_length_m/1000,1))
+# Clean up unnecessary objects
```

Comment:
ya dog. now we are cooking.  this is sweet but be careful.  seems like you are on it so I love it but don't forget when things change - its easy to forget you tidied up.